### ISSUE TITLE
[DatePicker] Pressing enter removes focus from DatePickerInput

### ISSUE THREAD

**Issue by diego-codes** [carbon-design-system/carbon#3211]
## Environment

> Operating system
OSX

> Browser
Chrome

## Detailed description

> What version of the Carbon Design System are you using?
`carbon-components-react@7.3.2`

> What did you expect to happen?
When manually typing a date in a `DatePickerInput`, the `DatePicker` requires that user presses the <kbd>Enter</kbd> key for it to register the new date, display it on the calendar UI, and call the `onChange` event. Pressing the <kbd>Enter</kbd> should maintain the keyboard focus on the `DatePickerInput`.

> What happened instead?
Pressing the <kbd>Enter</kbd> to register a manually entered date in the `DatePickerInput` removes the focus from the input and it goes back to the body.

> What WCAG 2.1 checkpoint does the issue violate?
Possibly:
- [2.4.7 Focus Visible](https://www.w3.org/TR/WCAG21/#focus-visible)
- [3.2.2 On Input](https://www.w3.org/TR/WCAG21/#on-input)
## Steps to reproduce the issue

1. Go to the DatePicker single with calendar [story](http://react.carbondesignsystem.com/?path=/story/datepicker--single-with-calendar)
2. Type a valid date value like `12/10/2018`
3. Press <kbd>Enter</kbd>
4. Notice that the keyboard focus is removed from the date picker altogether

## Additional information
![Kapture 2019-06-27 at 14 45 08](https://user-images.githubusercontent.com/5973294/60295786-31fdda00-98ea-11e9-81ed-d924d908cce3.gif)

**Comment by diego-codes** [comment]
Has this been identified as an accessibility issue?

**Comment by asudoh** [comment]
Seems that Flatpickr deliberately does that - Reported it at flatpickr/flatpickr#1895.

**Comment by emyarod** [comment]
I'm not able to reproduce this issue in the latest deploy preview

**Comment by asudoh** [comment]
I cannot reproduce it at react.carbondesignsystem.com, either.
